examples/onnx/yolov5-videos/predictor.py

```diff
@@ -0,0 +1,178 @@
+    )  # line/font thickness
+
+    for class_id, (x1, y1, x2, y2) in zip(class_ids, boxes.astype(np.int)):
+        color = color_map[class_id]
+        label = label_map[class_id]
+        cv2.rectangle(frame, (x1, y1), (x2, y2), color, tl, cv2.LINE_AA)
+        tf = max(tl - 1, 1)  # font thickness
+        t_size = cv2.getTextSize(label, 0, fontScale=tl / 3, thickness=tf)[0]
+        x3, y3 = x1 + t_size[0], y1 - t_size[1] - 3
+        cv2.rectangle(frame, (x1, y1), (x3, y3), color, -1, cv2.LINE_AA)  # filled
+        cv2.putText(
+            frame,
+            label,
+            (x1, y1 - 2),
+            0,
+            tl / 3,
+            [225, 255, 255],
+            thickness=tf,
+            lineType=cv2.LINE_AA,
+        )
+
+
+class ONNXPredictor:
+    def __init__(self, onnx_client, config):
+        self.client = onnx_client
+        # Get the input shape from the ONNX runtime
+        (signature,) = onnx_client.input_signatures.values()
+        _, _, height, width = signature["images"]["shape"]
+        self.input_size = (width, height)
+        self.config = config
+        with open("labels.json") as buf:
+            self.labels = json.load(buf)
+        color_map = plt.cm.tab20(np.linspace(0, 20, len(self.labels)))
+        self.color_map = [tuple(map(int, colors)) for colors in 255 * color_map]
+
+    def postprocess(self, output):
+        boxes, obj_score, class_scores = np.split(output[0], [4, 5], axis=1)
+        boxes = boxes_yolo_to_xyxy(boxes)
+
+        # get the class-prediction & class confidences
+        class_id = class_scores.argmax(axis=1)
+        cls_score = class_scores[np.arange(len(class_scores)), class_id]
+
+        confidence = obj_score.squeeze(axis=1) * cls_score
+        sel = confidence > self.config["confidence_threshold"]
+        boxes, class_id, confidence = boxes[sel], class_id[sel], confidence[sel]
+        sel = nms(boxes, confidence, self.config["iou_threshold"])
+        boxes, class_id, confidence = boxes[sel], class_id[sel], confidence[sel]
+        return boxes, class_id, confidence
+
+    def predict(self, payload):
+        in_path = download_from_youtube(payload["url"], self.input_size[1])
+        out_path = f"{uuid.uuid1()}.mp4"
+        writer = FrameWriter(out_path, size=self.input_size)
+
+        for frame in frame_reader(in_path, size=self.input_size):
+            x = (frame.astype(np.float32) / 255).transpose(2, 0, 1)
+            # 4 output tensors, the last three are intermediate values and
+            # not necessary for detection
+            output, *_ = self.client.predict(x[None])
+            boxes, class_ids, confidence = self.postprocess(output)
+            overlay_boxes(frame, boxes, class_ids, self.labels, self.color_map)
+            writer.write(frame)
+
+        del writer
+        os.remove(in_path)
+        # We cant remove out_path, so the deployment will run out of memory
```

Comment:
Will this cause the system to run out of memory or disk space?

Reply:
Disk space fortunately. #954 would fix it since we could remove the file there. I also considered using a combination of `BytesIO` and `StreamingResponse` to load the file into memory and delete it from disk. But since this is meant as an example, I think I prefer the current simpler version